src/test/java/com/github/davidmoten/rx2/aws/SqsTest.java

```diff
@@ -292,63 +314,129 @@ public void testPollingReturnsAllAvailableMessagesAtEachScheduledCall() {
-    // String queueName = bucketName;
-    // Sqs.queueName(queueName) //
-    // .sqsFactory(sqs) //
-    // .bucketName(bucketName) //
-    // .s3Factory(s3) //
-    // .messages() //
-    // .subscribeOn(Schedulers.io()) //
-    // .doOnNext(System.out::println) //
-    // .doOnNext(SqsMessage::deleteMessage) //
-    // .doOnError(e -> {
-    // e.printStackTrace();
-    // System.out.println(Thread.currentThread().getName());
-    // }) //
-    // .retryWhen(RetryWhen.delay(5, TimeUnit.SECONDS).build(), Schedulers.io())
-    // //
-    // .toBlocking().subscribe();
-    // }
 
+    @Test
+    public void testPollEvents() {
+        List<Message> list = new ArrayList<>();
+        List<String> events = new ArrayList<>();
+        List<Message> result = Sqs.messages(() -> list, () -> events.add("prePoll"), e -> events.add("postPoll"));
+        assertTrue(result == list);
+        assertEquals(Arrays.asList("prePoll", "postPoll"), events);
+    }
+    
+    @Test
+    public void testPollEventsIfPrePollThrows() {
+        List<Message> list = new ArrayList<>();
+        List<String> events = new ArrayList<>();
+        try {
+            Sqs.messages(() -> list, () -> {
+                throw new RuntimeException();
+            }, e -> events.add("postPoll"));
+            Assert.fail();
+        } catch (RuntimeException e) {
+            // expected
+            assertTrue(events.isEmpty());
+        }
+    }
+    
+    @Test
+    public void testPollEventsIfSupplierThrows() {
+        List<String> events = new ArrayList<>();
+        try {
+            Sqs.messages(() -> {
+                throw new RuntimeException();
+            }, () -> events.add("prePoll"), //
+                    e -> events.add(e.get().getClass().getSimpleName()));
+            Assert.fail();
+        } catch (RuntimeException e) {
+            // expected
+            assertEquals(Arrays.asList("prePoll", "RuntimeException"), events);
+        }
+    }
+    
+    @Test
+    public void testPollEventsIfPostPollThrows() {
+        List<String> events = new ArrayList<>();
+        try {
+            Sqs.messages(() -> new ArrayList<>(), //
+                    () -> events.add("prePoll"), //
+                    e -> {
+                        throw new IllegalArgumentException();
+                    });
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            // expected
+            assertEquals(Arrays.asList("prePoll"), events);
+        }
+    }
+    
+    @Test
+    public void testUncheckedCall() {
+        assertEquals(1, (int) Sqs.uncheckedCall(() -> 1));   
+    }
+    
+    @Test
+    public void testUncheckedCallThrowsRuntimeException() {
+        try {
+            Sqs.uncheckedCall(() -> {
+                throw new IllegalArgumentException("boo");
+            });
+            Assert.fail();
+        } catch (RuntimeException e) {
+            assertEquals("boo", e.getMessage());
```

Comment:
Same again - @Test(expected = RuntimeException.class)

Reply:
this pattern is used because another assertion happens after exception thrown